File: test/language/templates/cpp/InstantiateSimpleTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_simple/InstantiateSimple.h"

namespace templates
{
namespace instantiate_simple
{

TEST(InstantiateSimpleTest, readWrite)
{
    InstantiateSimple instantiateSimple;
    instantiateSimple.setTest(U32{13});

    zserio::BitStreamWriter writer;
    instantiateSimple.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    InstantiateSimple readInstantiateSimple(reader);

    ASSERT_TRUE(instantiateSimple == readInstantiateSimple);
}

} // namespace instantiate_simple
} // namespace templates

File: test/language/templates/python/InstantiateSimpleTest.py
import zserio

import Templates

class InstantiateSimpleTest(Templates.TestCase):
    def testReadWrite(self):
        instantiateSimple = self.api.InstantiateSimple(self.api.U32(13))

        writer = zserio.BitStreamWriter()
        instantiateSimple.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readInstantiateSimple = self.api.InstantiateSimple()
        readInstantiateSimple.read(reader)
        self.assertEqual(instantiateSimple, readInstantiateSimple)
